source/common/config/datasource.cc

```diff
@@ -77,6 +77,10 @@ absl::StatusOr<std::string> read(const envoy::config::core::v3::DataSource& sour
           fmt::format("Unexpected DataSource::specifier_case(): {}", source.specifier_case()));
     }
   }
+  if (max_size > 0 && data.length() > max_size) {
+    return absl::InvalidArgumentError(
+        fmt::format("response body size is {} bytes; maximum is {}", data.length(), max_size));
+  }
```

Comment:
This actually change the behavior of `max_direct_response_body_size_bytes` API of `RouteConfiguration`.In the new implementation, if 0 is set explicitly, then the file size would be unlimited.
**But considering the memory overhead, we have no plan for now to support unlimited file size.**

 Please remove this change. Thanks.

Reply:
With out this https://github.com/envoyproxy/envoy/blob/main/test/common/router/config_impl_test.cc#L7530 this test fails because the existing code https://github.com/envoyproxy/envoy/blob/84653b185c5c819a87f04bf4078630c2632a7671/source/common/router/config_utility.cc#L122 validates this. We do not have equivalent validation in the new implementation

Reply:
Rather than change the exist behavior, you can add the validation in the `DataSourceProvider::create`. (Or you can add a runtime flag to guard this behavior change).

But anyway, please remove the `max_size > 0` check, because we have no plan to use 0 to represent unlimited size. :)

Reply:
ok. Moved to create function